vortex-array/src/stats/stats_set.rs

```diff
@@ -427,6 +429,37 @@ impl StatsSet {
         }
     }
 
+    fn merge_sum(&mut self, other: &Self, dtype: &DType) {
+        match (
+            self.get_scalar_bound::<Sum>(dtype.clone()),
+            other.get_scalar_bound::<Sum>(dtype.clone()),
+        ) {
+            (Some(m1), Some(m2)) => {
+                // If the combine sum is exact, then we can sum them.
+                if let Some(scalar_value) = m1.zip(m2).some_exact().and_then(|(s1, s2)| {
+                    s1.as_primitive()
+                        .checked_add(&s2.as_primitive())
+                        .map(|pscalar| {
+                            pscalar
+                                .pvalue()
+                                .map(|pvalue| {
+                                    Scalar::primitive_value(
```

Comment:
Do we want to drop inexact sums, I think we shouldn't, Can do zip and + without the some_exact.